platform/security/sts/security-sts-certificateclaimshandler/src/test/java/org/codice/ddf/security/claims/certificate/CertificateClaimsHandlerTest.java

```diff
@@ -0,0 +1,195 @@
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.claims.certificate;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.cxf.rt.security.claims.Claim;
+import org.apache.cxf.rt.security.claims.ClaimCollection;
+import org.apache.cxf.sts.claims.ClaimsParameters;
+import org.apache.cxf.sts.claims.ProcessedClaimCollection;
+import org.junit.Test;
+
+import ddf.security.SubjectUtils;
+
+public class CertificateClaimsHandlerTest {
+
+    @Test
+    public void testGetSupportedClaimTypes() throws URISyntaxException {
+        CertificateClaimsHandler certificateClaimsHandler = new CertificateClaimsHandler();
+        List<URI> supportedClaimTypes = certificateClaimsHandler.getSupportedClaimTypes();
+        assertThat(supportedClaimTypes.size(), is(2));
+    }
+
+    @Test
+    public void testRetrieveClaimValuesNoCertValues() throws URISyntaxException {
+        CertificateClaimsHandler certificateClaimsHandler = new CertificateClaimsHandler();
+        ClaimCollection claims = new ClaimCollection();
+        Claim claim = new Claim();
+        claim.setClaimType(new URI(SubjectUtils.COUNTRY_CLAIM_URI));
+        claims.add(claim);
+        Claim claim1 = new Claim();
+        claim1.setClaimType(new URI(SubjectUtils.EMAIL_ADDRESS_CLAIM_URI));
+        claims.add(claim1);
+        ClaimsParameters parameters = new ClaimsParameters();
+        parameters.setPrincipal(mock(Principal.class));
+        ProcessedClaimCollection processedClaims = certificateClaimsHandler.retrieveClaimValues(
+                claims,
+                parameters);
+        assertThat(processedClaims.size(), is(0));
+    }
+
+    @Test
+    public void testRetrieveClaimValuesWithCertValues() throws URISyntaxException {
+        CertificateClaimsHandler certificateClaimsHandler = new CertificateClaimsHandler();
+        ClaimCollection claims = new ClaimCollection();
+        Claim claim = new Claim();
+        claim.setClaimType(new URI(SubjectUtils.COUNTRY_CLAIM_URI));
+        claims.add(claim);
+        Claim claim1 = new Claim();
+        claim1.setClaimType(new URI(SubjectUtils.EMAIL_ADDRESS_CLAIM_URI));
+        claims.add(claim1);
+        ClaimsParameters parameters = new ClaimsParameters();
```

Comment:
Could extract the default claims parameters into a helper method too since it is duplicated 3 times